include/deal.II/meshworker/assemble_flags.h

```diff
@@ -0,0 +1,209 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__assemble_flags_h
+#define dealii__assemble_flags_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/table.h>
+#include <deal.II/base/derivative_form.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/tensor.h>
+
+#include <vector>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/*!@addtogroup MeshWorker */
+/*@{*/
+
+/**
+ * The enum type given to the MeshLoop function, telling that function which
+ * elements need to be assembled.
+ *
+ * You can select more than one flag by concatenation using the bitwise or
+ * operator|(AssembleFlags,AssembleFlags).
+ *
+ * <h3>Use of these flags flags</h3>
+ *
+ * More information on the use of this type both in user code as well as
+ * internally can be found in the documentation modules on
+ * @ref AssembleFlags "The interplay of AssembleFlags, Mapping, and FiniteElement in FEValues"
+ * and
+ * @ref FE_vs_Mapping_vs_FEValues "How Mapping, FiniteElement, and FEValues work together".
+ */
+enum AssembleFlags
```

Comment:
do you want this to be in the `dealii` namespace, or in the `MeshWorker` namespace?